plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -817,4 +922,202 @@ private Map<String, String> getTableProperties(String schemaName, String tableNa
             return properties.buildOrThrow();
         }
     }
+
+    @Test
+    @Override
+    public void testSetColumnType()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
+        String tableName = "test_set_column_type";
+        assertUpdate("CREATE TABLE " + tableName + " (a bigint, b double NOT NULL, c varchar(50)) WITH (order_by=ARRAY['b'], engine = 'MergeTree')");
+        assertUpdate("ALTER TABLE " + tableName + " ALTER COLUMN a SET DATA TYPE varchar(50)");
+        assertEquals(getColumnType(tableName, "a"), "varchar");
+        assertThat((String) computeScalar("show create table " + tableName)).contains("CREATE TABLE clickhouse.tpch.test_set_column_type (\n" +
+                "   a varchar NOT NULL,\n" +
+                "   b double NOT NULL,\n" +
+                "   c varchar\n" +
+                ")\n" +
+                "WITH (\n" +
+                "   engine = 'MERGETREE',\n" +
+                "   order_by = ARRAY['b'],\n" +
+                "   primary_key = ARRAY['b']\n" +
+                ")");
+    }
+
+    @Test
+    @Override
+    public void testSetColumnIncompatibleType()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_SET_COLUMN_TYPE) && hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
+
+        String tableName = "test_set_column_incompatible_type";
+
+        assertUpdate("CREATE TABLE " + tableName + " (col bigint, col2 int not null) WITH (order_by=ARRAY['col2'], engine = 'MergeTree')");
+        assertUpdate("ALTER TABLE " + tableName + " ALTER COLUMN col SET DATA TYPE integer");
```

Comment:
The base test changes varchar type to integer. This overridden test is different from the base test purpose.

Reply:
How ClickHouse handle values which can't cast to integer? e.g. `test string`
Probably, we shouldn't allow changing the type from varchar to numeric types in the connector.